media-sound/supercollider/supercollider-3.8.0.ebuild

```diff
@@ -0,0 +1,116 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+# are all of theses needed ? 
+inherit cmake-utils eutils multilib versionator
+
+DESCRIPTION="An environment and a programming language for real time audio synthesis."
+HOMEPAGE="http://supercollider.github.io/"
+SRC_URI="https://github.com/supercollider/supercollider/releases/download/Version-${PV}/SuperCollider-${PV}-Source-linux.tar.bz2"
+
+RESTRICT="mirror"
+
+LICENSE="GPL-2 gpl3? ( GPL-3 )"
+SLOT="0"
+KEYWORDS="~x86 ~amd64"
+
+IUSE="
+	avahi curl debug doc emacs +fftw gedit +gpl3 ide jack portaudio qt5 server +sndfile
+	static-libs system-boost system-yaml vim wiimote
+	cpu_flags_x86_sse cpu_flags_x86_sse2"
+
+REQUIRED_USE="
+	ide? ( qt5 )
+	^^ ( jack portaudio )"
+
+# Both alsa and readline will be automatically checked in cmake but
+# there are no options for theese. Thus the functionality cannot be
+# controlled through USE flags. Therefore hard-enabled.
+RDEPEND="
+	portaudio? ( media-libs/portaudio )
+	jack? ( >=media-sound/jack2-1.9.10 )
+	media-libs/alsa-lib
+	>=sys-libs/readline-5.0:*
+	x11-libs/libXt
+	avahi? ( net-dns/avahi )
+	fftw? ( >=sci-libs/fftw-3.0 )
+	qt5? (
+		>=dev-qt/qtcore-5.6
+		>=dev-qt/qtwebkit-5.6
+		>=dev-qt/qtsensors-5.6
+		>=dev-qt/qtpositioning-5.6
+		>=dev-qt/qtgui-5.6
+	)
+	sndfile? ( >=media-libs/libsndfile-1.0.16 )
+	system-boost? ( dev-libs/boost )
+	system-yaml? ( dev-cpp/yaml-cpp )
+	wiimote? ( app-misc/cwiid )"
+
+DEPEND="${RDEPEND}
+	dev-libs/icu
+	virtual/pkgconfig
+	emacs? ( virtual/emacs )
+	gedit? ( app-editors/gedit )
+	vim? ( app-editors/vim )"
+
+S="${WORKDIR}/SuperCollider-Source"
+
+DOCS=( AUTHORS README_LINUX.md )
+
+src_configure() {
+	local mycmakeargs=(
+		-DNO_AVAHI=$(usex !avahi)
+		-DCURL=$(usex curl)
+		-DINSTALL_HELP=$(usex doc)
+		-DFFT_GREEN=$(usex !fftw)
+		-DNO_GPL3=$(usex !gpl3)
+		-DNO_SNDFILE=$(usex !sndfile)
+		-DSC_QT=$(usex qt5)
+		-DSCLANG_SERVER=$(usex server)
+		-DLIBSCSYNTH=$(usex !static-libs)
+		-DSYSTEM_BOOST=$(usex system-boost)
+		-DSYSTEM_YAMLCPP=$(usex system-yaml)
+		-DSSE=$(usex cpu_flags_x86_sse)
+		-DSSE2=$(usex cpu_flags_x86_sse2)
+		-DSC_IDE=$(usex ide)
+		-DSC_ED=$(usex gedit)
+		-DSC_VIM=$(usex vim)
+		-DSC_EL=$(usex emacs)
+		-DSC_WII=$(usex wiimote)
+	)
+	use jack && mycmakeargs+=( "AUDIOAPI=jack" )
+	use portaudio && mycmakeargs+=( "AUDIOAPI=portaudio" )
```

Comment:
seeing that you have a `^^ REQUIRED_USE`, just make this `AUDIOAPI=$(usex jack jack portaudio)`